### ISSUE TITLE
async in describe fails silently by not running tests

### ISSUE THREAD

**Issue by darvid7** [jasmine/jasmine#1734]
## Are you creating an issue in the correct repository?

- When in doubt, create an issue here.
- If you have an issue with the Jasmine docs, file an issue in the docs repo
  here: https://github.com/jasmine/jasmine.github.io
- This repository is for the core Jasmine framework
- If you are using a test runner that wraps Jasmine, consider filing an issue with that library if appropriate:
  - [Jasmine npm](https://github.com/jasmine/jasmine-npm/issues)
  - [Jasmine gem](https://github.com/jasmine/jasmine-gem/issues)
  - [Jasmine py](https://github.com/jasmine/jasmine-py/issues)
  - [Gulp Jasmine Browser](https://github.com/jasmine/gulp-jasmine-browser/issues)
  - [Karma](https://github.com/karma-runner/karma/issues)
  - [Grunt Contrib Jasmine](https://github.com/gruntjs/grunt-contrib-jasmine/issues)

So it looks like you can't do something like (well you can but the test won't run). However it doesn't tell you this describe block did not run or sometimes the text from the describe block is missing.
```
describe('my test', async () => {
await somePromise();
.....
});
```
## Expected Behavior

An error saying the set of tests did not run/warning saying something about async in the describe block.

## Current Behavior

Tests pass, the text in the describe block is omitted (and sometimes entire suites as well). 
All other tests pass so I thought all tests in my suite have passed, however the current test file I was working on was skipped. 

For example, in the below test suite the text `async describe with promise` is missing.
## Possible Solution

A meaningful error.

## Suite that reproduces the behavior (for bugs)

```javascript
import 'jasmine';

describe('testing stuff', () => {
  describe('async describe', async () => {
    it('computes a thing when asked', () => {
      expect(true).toBe(true);
    });
  });

  describe('async describe with promise', async () => {
    const yourNumber = await new Promise((resolve, reject) => {
      resolve(10);
    });

    it('computes a thing when asked', () => {
      expect(yourNumber).toBe(10);
      console.log(`your number here ${yourNumber}`);
      expect(true).toBe(true);
    });
  });

  describe('async describe with declaration', async () => {
    let myNumber: number;
    it('computes a thing when asked', async () => {
      myNumber = await new Promise((resolve, reject) => {
        resolve(10);
      });
      expect(myNumber).toBe(10);
      expect(true).toBe(true);
    });
  })
});
```
## Your Environment

* Version used:
* Environment name and version (e.g. Chrome 39, node.js 5.4):
* Operating System and version (desktop or mobile):
* Link to your project:

**Comment by slackersoft** [comment]
What version of Jasmine are you using? Newer versions of Jasmine (> 2.6) should actually throw an error when a `describe` is passed something that is not a bare `function` (See #1222).

**Comment by sgravrock** [comment]
I ran the example that you provided in Jasmine 3.5 and got the following output, which I believe is what you're asking for:

```
[...]
Suite error: testing stuff
  Message:
    Error: describe expects a function argument; received [object AsyncFunction]
[...]
```

I'm going to go ahead and close this since it looks like it was fixed in 2.6 and we haven't heard anything more in quite some time. Feel free to reopen and provide more information if you're still able to reproduce this with the latest version of Jasmine.
